src/Entity/OgRole.php

```diff
@@ -177,8 +177,39 @@ public function setRoleType($role_type) {
   /**
    * {@inheritdoc}
    */
+  public function getName() {
+    // If the name is not set yet, try to derive it from the ID.
+    if (empty($this->name) && !empty($this->id())) {
+      list(, , $name) = explode('-', $this->id());
+      $this->setName($name);
+    }
+    return $this->name;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setName($name) {
+    $this->name = $name;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function save() {
-    if ($this->isNew()) {
+    // The ID of a new OgRole has to consist of the entity type ID, bundle ID
```

Comment:
@damiankloip if you want to provide a role that is available for all groups, you can provide an event subscriber that simply ignores the passed in group entity type and bundle and returns the role every time, eg:

```
class MyEventSubscriber implements EventSubscriberInterface {

  public static function getSubscribedEvents() {
    return [
      DefaultRoleEventInterface::EVENT_NAME => [['provideDefaultRoles']],
    ];
  }

  public function provideDefaultRoles(DefaultRoleEventInterface $event) {
    // Just ignore $event->getEntityTypeId() and $event->getBundleId(),
    // this role applies to all groups.
    $event->addRole([
      'name' => 'moderator',
      'label' => 'Moderator',
      'is_admin' => TRUE,
    ]);
  }

}
```